generator/feature_builder.hpp

```diff
@@ -187,146 +112,146 @@ class FeatureBuilder1
     return ForAnyGeometryPointEx(std::move(wrapper));
   }
 
-  bool PreSerialize();
+  // To work with geometry type.
+  void SetCenter(m2::PointD const & p);
+  void SetLinear(bool reverseGeometry = false);
+  void SetArea() { m_params.SetGeomType(GeomType::Area); }
+  bool IsPoint() const { return GetGeomType() == GeomType::Point; }
+  bool IsLine() const { return GetGeomType() == GeomType::Line; }
+  bool IsArea() const { return GetGeomType() == GeomType::Area; }
 
-  bool PreSerializeAndRemoveUselessNames();
+  // To work with types.
+  void SetType(uint32_t type) { m_params.SetType(type); }
+  void AddType(uint32_t type) { m_params.AddType(type); }
+  bool PopExactType(uint32_t type) { return m_params.PopExactType(type); }
+  template <class FnT>
```

Comment:
empty line before and after method templates, please
s/FnT/Fn/

Reply:
++